### PR TITLE
Change V3.1

### LINKED ISSUE #299: File.value / PathType is inconsistent / too strict
PathType is defined on page 97 as:

Identifier
Note: for any string conformant to RFC8089, the "file" URI scheme (for relative and absolute file paths) applies.

Exmaples:
./Specification.pdf
file:c:/local/Specification.pdf
file://host.example.com/path/to/file

This is more strict than it was in Version 2 (it was a simple string) and inconsistent, since the example ./Specification.pdf is not valid according to RFC8089.

As discussed in Workstream AAS, the specification should be changed to:

`URL, based on RFC-3986 / RFC-3987, but more recently defined as living standard by WHATWG, as suggested by W3C: https://url.spec.whatwg.org/`

In the examples we should also list http / ftp  / custom-scheme examples - since all of them are considered valid.

### LINKED ISSUE #361: Spec. Part 1 (01001-3-0): Make clear that the semanticId defines the semantics (label) for RelationshipElement
Under 5.3.7.15 it is not stated as in 5.3.7.2 (AnnotatedRelationShipElement) that the semanticId of the RelationshipElement holds the semantic (label) of the relation. 

It should be stated under 5.3.7.15 as well.

### PR DESCRIPTION
**Pull request by BirgitBoss**
This issue closes 
https://github.com/admin-shell-io/aas-specs/issues/299 
shall be anyUri instead of string + length restriction and note

add anyUri to basic data types section
change PathType definition
update ChangeLog

closes #361

### CODE REVIEW

**Comment by mhrimaz** [comment]
@BirgitBoss 
Sorry that I jump in, but would that be possible to also change the datatype of other values that are now 'xs:string' to their actual XSD data type?
I have elaborated the issue here: https://github.com/admin-shell-io/aas-specs/issues/284
The problem right now is that if we have a `Property` with a `xs:double` valueType, in RDF we use string.

So currently we have:
```turtle
<something_48c66017> rdf:type aas:Submodel ;
    <https://admin-shell.io/aas/3/0/Identifiable/id> "something_48c66017"^^xs:string ;
    <https://admin-shell.io/aas/3/0/Submodel/submodelElements> [
        rdf:type aas:Property ;
        <https://admin-shell.io/aas/3/0/Referable/idShort> "something3fdd3eb4"^^xs:string ;
        <https://admin-shell.io/aas/3/0/Property/valueType> <https://admin-shell.io/aas/3/0/DataTypeDefXsd/Double> ;
###
        <https://admin-shell.io/aas/3/0/Property/value> "1234.01234"^^xs:string ;
    ] ;
.
```

but it should be:

```turtle
<something_48c66017> rdf:type aas:Submodel ;
    <https://admin-shell.io/aas/3/0/Identifiable/id> "something_48c66017"^^xs:string ;
    <https://admin-shell.io/aas/3/0/Submodel/submodelElements> [
        rdf:type aas:Property ;
        <https://admin-shell.io/aas/3/0/Referable/idShort> "something3fdd3eb4"^^xs:string ;
        <https://admin-shell.io/aas/3/0/Property/valueType> <https://admin-shell.io/aas/3/0/DataTypeDefXsd/Double> ;
###
        <https://admin-shell.io/aas/3/0/Property/value> "1234.01234"^^xs:double;
    ] ;
.
```

It might be a breaking change, but it doesn't cause any issues for roundtrips, since the metamodel anyways use XSD datatypes. This would allow to write SPARQL query easier (numeric aggeragation, comparison, ...)

**Comment by BirgitBoss** [comment]
> @BirgitBoss Sorry that I jump in, but would that be possible to also change the datatype of other values that are now 'xs:string' to their actual XSD data type? I have elaborated the issue here: #284 

Thank you for pointing to this issue. They are both about data types but your problem is more tricky because the type of the property is not always the same but depends on its semantic definition and implementation. For the SME "File" the type is static. So I prefer to discuss the typing issue of Properties (and similar SME) in #284 and to discuss only data type "PathType" in this issue.

Thank you for your understanding.

File: documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc [hunk 1]
```diff
@@ -12,24 +12,39 @@ SPDX-License-Identifier: CC-BY-4.0
...
 |Source |Basic Data Type |Value Range |Sample Values
+
 |xsd |string |Character string (but not all Unicode character strings) |"Hello world", "Καλημέρα κόσμε", +
 "ハローワールド""
+
 |xsd |base64Binary |base64-encoded binary data |"a3Vtb3dhc2hlcmU="
+
 |xsd |boolean |true, false |true, false
+
+|xsd |anyUri | Represents an Internationalized Resource Identifier Reference (IRI) conformant to https://www.ietf.org/rfc/rfc3987.txt[RFC 3987], can be absolute or relative, and may have an optional fragment identifier  a| {blank}./Specification.pdf
```
Comment (alexgordtop) [inline documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc:34]: This is inconsistent to line 203 - where we write:

...as URIs per [RFC 3986] and its successor(s).

Please replace with:
|xsd |anyUri | Represents an Uniform Resource Identifier (URI) conformant to https://www.ietf.org/rfc/rfc3986[RFC 3986] and its successor(s), can be absolute or relative, and may have an optional fragment identifier  a| {blank}./Specification.pdf

**Review by alexgordtop — CHANGES_REQUESTED** [review]
I think we should stick with RFC3986, which defines URIs, that are required to resolve resources. This also addresses the issue of fishing attacks with character symbols, that look like the original ones, but lead to a different address. URIs don't have this issue, while IRIs do.

File: documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc (hunk 1 above)
Comment (BirgitBoss) [reply documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc]: You are right. Resolved with https://github.com/admin-shell-io/aas-specs/pull/358/commits/d59781308ee906a9e37533adeef4b1408ff65c9f

**Comment by BirgitBoss** [comment]
@alexgordtop can you now approve of the PR after the changes made?

**Review by alexgordtop — APPROVED** [review]

**Comment by g1zzm0** [comment]
**[PathType ](https://aas-core-works.github.io/aas-core-meta/v3_1/PathType.html) should not change its type!**

Rather, it should migrate the constraints from [RFC8089 ](https://datatracker.ietf.org/doc/html/rfc8089) to [RFC3987](https://datatracker.ietf.org/doc/html/rfc3987). This could be achieved by changing from [matchesRFC8089 ](https://aas-core-works.github.io/aas-core-meta/v3_1/MatchesRfc8089Path.html) to [matchesXsAnyUri](https://aas-core-works.github.io/aas-core-meta/v3_1/MatchesXsAnyUri.html). This is necessary to avoid having to create an entire new RegEx from @s-heppner or @mristin because of time constraints. The task force wanted the change from RFC8089 to RFC3987, but nobody wanted to make the change.

According to version 3.0, [PathType](https://aas-core-works.github.io/aas-core-meta/v3/PathType.html#inheritances) is an [Identifier](https://aas-core-works.github.io/aas-core-meta/v3/Identifier.html) that is also subject to the matchesRFC8089 constraint. The PathType cannot simply be changed to anyUri because **anyUri is not a primitive type**. This is breaking to version 3.0 and cannot be implemented in aas-core. 
**No approval from me.**

File: documentation/IDTA-01001/modules/ROOT/pages/IDTA-01001_ChangeLog.adoc [hunk 2]
```diff
@@ -103,7 +103,7 @@ Bugfixes:
 | | Identifier a| data type: change length from 2000 to 2024 characters
 | | MultiLanguageNameType a| (bugfix in Text only) maxLength=128 (as derived and implemented)
 | | NameType a| implicit change due to change of Constraint AASd-002
-| | PathType a| extend data type to support complete URI scheme for files RFC3986/RFC3987
+| | PathType a| type changed from string to anyUri, max length 2024
```
Comment (g1zzm0) [inline documentation/IDTA-01001/modules/ROOT/pages/IDTA-01001_ChangeLog.adoc:106]:
```suggestion
| | PathType a| extend data type to support complete URI scheme for files RFC 2396/RFC 2732
```

File: documentation/IDTA-01001/modules/ROOT/pages/IDTA-01001_ChangeLog.adoc [hunk 3]
```diff
@@ -46,7 +46,7 @@ Major Changes:
 
 
 * New value "Role" to enumeration AssetKind (https://github.com/admin-shell-io/aas-specs/issues/294[#294])
-* PathType: support complete URI scheme for files RFC3986/RFC3987 (https://github.com/admin-shell-io/aas-specs/issues/299[#299])
+* Change PathType: type changed from string to anyUri, length restriction 2024. Reason is to support relative path and other URL schemes (https://github.com/admin-shell-io/aas-specs/issues/299[#299])
```
Comment (g1zzm0) [inline documentation/IDTA-01001/modules/ROOT/pages/IDTA-01001_ChangeLog.adoc:49]:
```suggestion
* Change PathType: support complete URI scheme for files RFC2396/RFC2732  (https://github.com/admin-shell-io/aas-specs/issues/299[#299])
```

File: documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc [hunk 4]
```diff
@@ -12,24 +12,41 @@ SPDX-License-Identifier: CC-BY-4.0
 
 == Predefined Simple Data Types
 
-The metamodel of the Asset Administration Shell uses basic data types as defined in the link:https://www.w3.org/XML/Core/[XML Schema Definition] (XSD)footnote:[https://www.w3.org/XML/Core/, former https://www.w3.org/XML/Schema]. See Table 7 for an overview of the used types. Their definition is outside the scope of this document.
+The metamodel of the Asset Administration Shell uses some of the predefined simple data types of the  link:https://www.w3.org/XML/Core/[XML Schema Definition] (XSD) 
+as its basic data types footnote:[https://www.w3.org/XML/Core/, former https://www.w3.org/XML/Schema]. See Table 7 for an overview of the used types. Their definition is outside the scope of this document.
```
Comment (g1zzm0) [inline documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc:16]:
```suggestion
as its basic data types footnote:[https://www.w3.org/XML/Core/, former https://www.w3.org/TR/xmlschema-2/#built-in-datatypes]. See Table 7 for an overview of the used types. Their definition is outside the scope of this document.
```

Comment (g1zzm0) [reply documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc:16]: https://www.w3.org/XML/Schema points directly to XML Schema 1.1

**Comment by BirgitBoss** [comment]
see revised decision in https://github.com/admin-shell-io/aas-specs/issues/299

File: documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc [hunk 5]
```diff
@@ -174,18 +191,25 @@ Each string within langString has a max of 1,023 and min of 1 characters.
...
+_anyURI_
+
+with max 2024 and min 1 characters
 
 
 ====
-Note: URL, based on [RFC-3986](https://www.rfc-editor.org/rfc/rfc3987) /[RFC-3987](https://www.rfc-editor.org/rfc/rfc3987), but more recently defined as living standard by WHATWG, as suggested by W3C: https://url.spec.whatwg.org/
+"[anyURI supports a] wide range of internationalized resource identifiers can be specified when an anyURI is called for, and still be understood as URIs per https://www.w3.org/TR/xmlschema11-2/#RFC3986[RFC 3986] and its successor(s)." 
+
+Source: https://www.w3.org/TR/xmlschema11-2/#anyURI[W3C XML Schema Definition Language (XSD) 1.1 Part 2: Datatypes]
```
Comment (g1zzm0) [inline documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc:207]:
```suggestion
Source: https://www.w3.org/TR/xmlschema-2/#anyURI[W3C XML Schema Definition Language (XSD) 1.0 Part 2: Datatypes]
```

File: documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc [hunk 6]
```diff
@@ -174,18 +191,25 @@ Each string within langString has a max of 1,023 and min of 1 characters.
...
-_Identifier_
+
+_anyURI_
+
+with max 2024 and min 1 characters
 
 
 ====
-Note: URL, based on [RFC-3986](https://www.rfc-editor.org/rfc/rfc3987) /[RFC-3987](https://www.rfc-editor.org/rfc/rfc3987), but more recently defined as living standard by WHATWG, as suggested by W3C: https://url.spec.whatwg.org/
+"[anyURI supports a] wide range of internationalized resource identifiers can be specified when an anyURI is called for, and still be understood as URIs per https://www.w3.org/TR/xmlschema11-2/#RFC3986[RFC 3986] and its successor(s)."
```
Comment (g1zzm0) [inline documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc:205]:
```suggestion
"[anyURI supports a] wide range of internationalized resource identifiers can be specified when an anyURI is called for, and still be understood as URIs per https://www.w3.org/TR/xmlschema-2/#RFC2396[RFC 2396] and https://www.w3.org/TR/xmlschema-2/#RFC2732[RFC 2732] ." 
```

File: documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc [hunk 7]
```diff
@@ -12,24 +12,41 @@ SPDX-License-Identifier: CC-BY-4.0
...
+
+\http://www.example.org
+
+FTP://unicode.org
+
+|xsd |https://www.w3.org/TR/xmlschema2/#dateType[dateType] |Date and time with or without time zone |"2000-01-01T14:23:00", +
 "2000-01-01T14:23:00.66372+14:00"footnote:[Corresponds to xs:dateTimeStamp in XML Schema 1.1]
-|xsd |duration |Duration of time |"-P1Y2M3DT1H", +
+
+|xsd |https://www.w3.org/TR/xmlschema2/#duration[duration] |Duration of time |"-P1Y2M3DT1H", +
```
Comment (g1zzm0) [inline documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc:47]:
```suggestion
|xsd |https://www.w3.org/TR/xmlschema-2/#duration[duration] |Duration of time |"-P1Y2M3DT1H", +
```

File: documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc [hunk 8]
```diff
@@ -12,24 +12,41 @@ SPDX-License-Identifier: CC-BY-4.0
...
+
+It can be absolute or relative, and may have an optional fragment identifier  a| {blank}./Specification.pdf
+
+file:c:/local/Specification.pdf
+
+\http://www.example.org
+
+FTP://unicode.org
+
+|xsd |https://www.w3.org/TR/xmlschema2/#dateType[dateType] |Date and time with or without time zone |"2000-01-01T14:23:00", +
```
Comment (g1zzm0) [inline documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc:44]:
```suggestion
|xsd |https://www.w3.org/TR/xmlschema-2/#d0e11239[dateType] |Date and time with or without time zone |"2000-01-01T14:23:00", +
```

Comment (g1zzm0) [inline documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc:44]:
```suggestion
|xsd |https://www.w3.org/TR/xmlschema-2/#d0e11239[dateType] |Date and time with or without time zone |"2000-01-01T14:23:00", +
```

Comment (g1zzm0) [reply documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc:44]:
```suggestion
|xsd |https://www.w3.org/TR/xmlschema-2/#dateTime[dateTime] |Date and time with or without time zone |"2000-01-01T14:23:00", +
```

File: documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc [hunk 9]
```diff
@@ -12,24 +12,41 @@ SPDX-License-Identifier: CC-BY-4.0
...
 |Source |Basic Data Type |Value Range |Sample Values
-|xsd |string |Character string (but not all Unicode character strings) |"Hello world", "Καλημέρα κόσμε", +
+
+|xsd |https://www.w3.org/TR/xmlschema2/#string[string] |Character string (but not all Unicode character strings) |"Hello world", "Καλημέρα κόσμε", +
 "ハローワールド""
-|xsd |base64Binary |base64-encoded binary data |"a3Vtb3dhc2hlcmU="
-|xsd |boolean |true, false |true, false
-|xsd |dateType |Date and time with or without time zone |"2000-01-01T14:23:00", +
+
+|xsd |https://www.w3.org/TR/xmlschema2/#base64Binary[base64Binary] |base64-encoded binary data |"a3Vtb3dhc2hlcmU="
```
Comment (g1zzm0) [inline documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc:30]:
```suggestion
|xsd |https://www.w3.org/TR/xmlschema-2/#base64Binary[base64Binary] |base64-encoded binary data |"a3Vtb3dhc2hlcmU="
```

File: documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc [hunk 10]
```diff
@@ -12,24 +12,41 @@ SPDX-License-Identifier: CC-BY-4.0
...
 
 See Clause 5.3.12.6 for constraints on types.
 
 .Simple Data Types Used in Metamodel
 [cols="10%,19%,38%,33%",options="header",]
 |===
 |Source |Basic Data Type |Value Range |Sample Values
-|xsd |string |Character string (but not all Unicode character strings) |"Hello world", "Καλημέρα κόσμε", +
+
+|xsd |https://www.w3.org/TR/xmlschema2/#string[string] |Character string (but not all Unicode character strings) |"Hello world", "Καλημέρα κόσμε", +
```
Comment (g1zzm0) [inline documentation/IDTA-01001/modules/ROOT/pages/Spec/IDTA-01001_Metamodel_DataTypes.adoc:27]:
```suggestion
|xsd |https://www.w3.org/TR/xmlschema-2/#string[string] |Character string (but not all Unicode character strings) |"Hello world", "Καλημέρα κόσμε", +
```

**Comment by g1zzm0** [comment]
Allways a good idea to mege unaproved PRs.
Please check twice if all XSD Spec Version 1.0 URLs point to https://www.w3.org/TR/xmlschema-2/ and not tohttps://www.w3.org/TR/xmlschema2/ and for the wrong URLs that point to the wrong XSD Version 1.1 with path https://www.w3.org/TR/xmlschema11-2/

**Pull request merged**